tests/foreman/api/test_contentmanagement.py

```diff
@@ -1463,3 +1463,32 @@ def test_positive_sync_file_repo(
             sat_file = md5_by_url(f'{sat_repo_url}{file}')
             caps_file = md5_by_url(f'{caps_repo_url}{file}')
             assert sat_file == caps_file
+
+    @pytest.mark.tier3
+    def test_positive_allow_reregistration_when_dmi_uuid_changed(
+        self, module_org, rhel7_contenthost, default_sat
+    ):
+        """Register a content host with a custom DMI UUID, unregistering it, change
+        the DMI UUID, and re-registering it again
+
+        :id: 7f431cb2-5a63-41f7-a27f-62b86328b50d
+
+        :expectedresults: The content host registers successfully
+
+        :BZ: 1747177
+
+        :CaseLevel: Integration
+        """
+        rhel7_contenthost.install_katello_ca(default_sat)
+        default_sat.execute(
+            'echo \'{{"dmi.system.uuid": "this-is-made-up"}}\' > /etc/rhsm/facts/uuid.facts'
+        )
+        result = rhel7_contenthost.register_contenthost(module_org.label, lce='Library')
+        assert result.status == 0
+        result = rhel7_contenthost.unregister()
+        assert result.status == 0
+        default_sat.execute(
+            'echo \'{{"dmi.system.uuid": "another-made-up"}}\' > /etc/rhsm/facts/uuid.facts'
```

Comment:
Consider using valid UUID (hex characters only) here

Reply:
@lhellebr Would using `gen_string('alpha)` but acceptable here to generate some UUID?  If not, what is?

Reply:
It may work even with invalid UUID. However, if you want a valid UUID here, you want something like that:
```
# subscription-manager facts | grep dmi.system.uuid
dmi.system.uuid: BE50E0DA-2EF2-4469-8EF5-17EC97E19710
```
`gen_alphanumeric` won't give you that. There are tons of articles on generating UUIDs in Python, one example:
```
>>> import uuid; print(f"Generated UUID: {uuid.uuid1()}")
Generated UUID: 6c589f10-ae7b-11ec-b274-566f0d721523
```